packages/openai/src/openai-speech-model.ts

```diff
@@ -0,0 +1,162 @@
+    ),
+  response_format: z
+    .enum(['mp3', 'opus', 'aac', 'flac', 'wav', 'pcm'])
+    .default('mp3')
+    .describe('The format of the generated audio.'),
+  speed: z
+    .number()
+    .min(0.25)
+    .max(4.0)
+    .default(1.0)
+    .describe('The speed of the generated audio.'),
+});
+
+export type OpenAISpeechCallOptions = Omit<
+  SpeechModelV1CallOptions,
+  'providerOptions'
+> & {
+  providerOptions?: {
+    openai?: z.infer<typeof OpenAIProviderOptionsSchema>;
+  };
+};
+
+interface OpenAISpeechModelConfig extends OpenAIConfig {
+  _internal?: {
+    currentDate?: () => Date;
+  };
+}
+
+export class OpenAISpeechModel implements SpeechModelV1 {
+  readonly specificationVersion = 'v1';
+
+  get provider(): string {
+    return this.config.provider;
+  }
+
+  constructor(
+    readonly modelId: OpenAISpeechModelId,
+    private readonly config: OpenAISpeechModelConfig,
+  ) {}
+
+  private getArgs({ text, providerOptions }: OpenAISpeechCallOptions) {
+    const warnings: SpeechModelV1CallWarning[] = [];
+
+    // Parse provider options
+    const openAIOptions = parseProviderOptions({
+      provider: 'openai',
+      providerOptions,
+      schema: OpenAIProviderOptionsSchema,
+    });
+
+    // Create request body
+    const requestBody: Record<string, any> = {
+      model: this.modelId,
+      input: text,
+      voice: 'alloy',
+    };
+
+    // Add provider-specific options
+    if (openAIOptions) {
+      const speechModelOptions: OpenAISpeechModelOptions = {
+        voice: openAIOptions.voice,
+        speed: openAIOptions.speed,
+        response_format: openAIOptions.response_format,
+        instructions: openAIOptions.instructions,
+      };
+
+      for (const key in speechModelOptions) {
+        const value = speechModelOptions[key as keyof OpenAISpeechModelOptions];
+        if (value !== undefined) {
+          requestBody[key] = value;
+        }
+      }
+    }
+
+    return {
+      requestBody,
+      warnings,
+    };
+  }
+
+  async doGenerate(
+    options: OpenAISpeechCallOptions,
+  ): Promise<Awaited<ReturnType<SpeechModelV1['doGenerate']>>> {
+    const currentDate = this.config._internal?.currentDate?.() ?? new Date();
+    const { requestBody, warnings } = this.getArgs(options);
+
+    const {
+      value: audioBuffer,
+      responseHeaders,
+      rawValue: rawResponse,
+    } = await postJsonToApi({
+      url: this.config.url({
+        path: '/audio/speech',
+        modelId: this.modelId,
+      }),
+      headers: combineHeaders(this.config.headers(), options.headers),
+      body: requestBody,
+      failedResponseHandler: openaiFailedResponseHandler,
+      successfulResponseHandler: createBinaryResponseHandler(),
+      abortSignal: options.abortSignal,
+      fetch: this.config.fetch,
+    });
+
+    return {
+      audioData: audioBuffer.buffer,
+      contentType: `audio/${options.providerOptions?.openai?.response_format ?? 'mp3'}`,
```

Comment:
hmm wondering if we should have this at all on the model level. we can just detect the content type in the `ai` lib